gradio/networking.py

```diff
@@ -148,20 +130,51 @@ def start_server(
 
     app = App.create_app(blocks, app_kwargs=app_kwargs)
 
-    if blocks.save_to is not None:  # Used for selenium tests
-        blocks.save_to["port"] = port
-    config = uvicorn.Config(
-        app=app,
-        port=port,
-        host=host,
-        log_level="warning",
-        ssl_keyfile=ssl_keyfile,
-        ssl_certfile=ssl_certfile,
-        ssl_keyfile_password=ssl_keyfile_password,
-        ws_max_size=1024 * 1024 * 1024,  # Setting max websocket size to be 1 GB
+    server_ports = (
+        [server_port]
+        if server_port is not None
+        else range(INITIAL_PORT_VALUE, INITIAL_PORT_VALUE + TRY_NUM_PORTS)
     )
-    server = Server(config=config)
-    server.run_in_thread()
+
+    for port in server_ports:
+        try:
+            # The fastest way to check if a port is available is to try to bind to it with socket.

```

Comment:
Sorry for the copious comments, but I wanted to make sure the new logic is clear to someone who is reading the code for the first time.

Reply:
I appreciate it!